swap_meet/vendor.py

```diff
@@ -1,2 +1,64 @@
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in set(self.inventory):
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+    
+    def get_by_category(self, category):
+        cat_list = []
+        for item in self.inventory:
+            if item.category == category:
+                cat_list.append(item)
+        return cat_list
+    
+    def swap_items(self, other, my_item, their_item):
+        try:
+            my_index = self.inventory.index(my_item)
+            their_index = other.inventory.index(their_item)
+            self.inventory[my_index], other.inventory[their_index] = other.inventory[their_index], self.inventory[my_index]
+            return True
+        except:
+            return False
+    
+    def swap_first_item(self, other):
+        if len(self.inventory) >= 1 and len(other.inventory) >= 1:
+            self.inventory[0], other.inventory[0] = other.inventory[0], self.inventory[0]
```

Comment:
Note that we could still use `swap_items` here and the time complexity would still effectively be constant. Since the items are in the firs position, `index` would find them in the first position and swap them there.